site/app/templates/admin/SystemUpdate.twig

```diff
@@ -0,0 +1,43 @@
+<div class="content">
+    <h2>System Update</h2>
+    <h4> Current Version Installed {{ latest_tag }} </h4>
+    <a target="_blank" href="https://submitty.org/sysadmin/update">Need help with updating Submitty? <i style="font-style:normal;" class="fa-question-circle"></i></a>
+    <button class="btn btn-primary" onclick="getReleases('{{latest_tag}}');">Check for Updates</button>
+    <div class="content" id="update_status" hidden>
+        <h2 id="text"></label>
+        <h4 id="tag"></label> 
+    </div>
+
+    <div class="content" id="release_notes_container" hidden>
+        <label id="release_notes"></label>
+    </div>
+</div>
+
+
+<script>
+    // "https://api.github.com/repos/Submitty/Submitty/releases"
+    function getReleases(latest_tag) {
+        var xmlHttp = new XMLHttpRequest();
+        xmlHttp.open( "GET", "https://api.github.com/repos/Submitty/Submitty/releases", false ); // false for synchronous request
+        xmlHttp.send( null );
+        //console.log(xmlHttp.responseText);
+        var myArr = JSON.parse(xmlHttp.responseText);
+        console.log(myArr[0]);
+        if(myArr[0]["tag_name"] == latest_tag) {
+            document.getElementById('text').innerHTML = "Submitty is up to date!";
+
+        } else {
+            document.getElementById('text').innerHTML = "A newer version of Submitty is available";
+            document.getElementById('tag').innerHTML = "Current Version Installed: " + latest_tag + "\n" + "Most Recent Version: " + myArr[0]["tag_name"];
+        }
+
+        document.getElementById('release_notes').innerHTML = parseReleaseNotes(myArr[0]["body"]);
+        document.getElementById('release_notes_container').style.display = "block"
+        document.getElementById('update_status').style.display = "block"
+    }
```

Comment:
```suggestion
    async function getReleases(latest_tag) {
        try {
            const req = await fetch('https://api.github.com/repos/Submitty/Submitty/releases');
            const release = await req.json()[0];
            if (release["tag_name"] === latest_tag) {
                document.getElementById('text').innerHTML = "Submitty is up to date!";
            }
            else {
                document.getElementById('text').innerHTML = "A newer version of Submitty is available";
                document.getElementById('tag').innerHTML = "Current Version Installed: " + latest_tag + "\n" + "Most Recent Version: " + release["tag_name"];
            }
        }
        catch (exc) {
            document.getElementById('text').innerHTML = 'Could not fetch version information';
        }

        document.getElementById('release_notes').innerHTML = parseReleaseNotes(release["body"]);
        document.getElementById('release_notes_container').style.display = "block"
        document.getElementById('update_status').style.display = "block"
    }
```

Reply:
Let's also trigger this function on page load, and not require instructor to press a button. If they're coming to this page, chances are they want to know version information and if update is available.

Reply:
So for some words on the above:
1. The [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) api is awesome and we should use that instead of jQuery or old school `XMLHttpRequest` as it's a sane, promise based interface
1. We should handle the case that we cannot fetch the version information for whatever reason (no internet, github is down, etc.)
1. Use `const` / `let` over `var` and avoid wacky scoping rules